detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformerv6.java

```diff
@@ -0,0 +1,201 @@
+            Optional<Dependency> pnpmPackage, Map.Entry<String, String> packageDependency) {
+        String dependencyPackageId = convertRawEntryToPackageId(packageDependency.getKey(), packageDependency.getValue(), linkedPackageResolver, reportingProjectPackagePath);
+        Optional<Dependency> child = buildDependencyFromPackageId(dependencyPackageId);
+        child.ifPresent(c -> graphBuilder.addChildWithParent(child.get(), pnpmPackage.get()));
+    }
+
+    private PnpmProjectPackagev6 convertPnpmLockYamlToPnpmProjectPackage(PnpmLockYamlv6 pnpmLockYaml) {
+        PnpmProjectPackagev6 pnpmProjectPackage = new PnpmProjectPackagev6();
+
+        pnpmProjectPackage.dependencies = pnpmLockYaml.dependencies;
+        pnpmProjectPackage.devDependencies = pnpmLockYaml.devDependencies;
+        pnpmProjectPackage.optionalDependencies = pnpmLockYaml.optionalDependencies;
+
+        return pnpmProjectPackage;
+    }
+
+    private List<String> extractRootPackageIds(
+        PnpmProjectPackagev6 pnpmProjectPackage,
+        @Nullable String reportingProjectPackagePath,
+        PnpmLinkedPackageResolver linkedPackageResolver
+    ) {
+        Map<String, PnpmDependencyInfo> rawPackageInfo = new HashMap<>();
+        if (pnpmProjectPackage.dependencies != null) {
+            rawPackageInfo.putAll(pnpmProjectPackage.dependencies);
+        }
+        dependencyTypeFilter.ifShouldInclude(PnpmDependencyType.DEV, pnpmProjectPackage.devDependencies, rawPackageInfo::putAll);
+        dependencyTypeFilter.ifShouldInclude(PnpmDependencyType.OPTIONAL, pnpmProjectPackage.optionalDependencies, rawPackageInfo::putAll);
+
+        return rawPackageInfo.entrySet().stream()
+            .map(entry -> convertRawEntryToPackageId(entry.getKey(), entry.getValue().version, linkedPackageResolver, reportingProjectPackagePath))
+            .collect(Collectors.toList());
+    }
+
+    private String convertRawEntryToPackageId(String name, String version, PnpmLinkedPackageResolver linkedPackageResolver, @Nullable String reportingProjectPackagePath) {
+        name = StringUtils.strip(name, "'");
+        if (version.startsWith(LINKED_PACKAGE_PREFIX)) {
+            // a linked project package's version will be referenced in the format: <linkPrefix><pathToLinkedPackageRelativeToReportingProjectPackage>
+            version = linkedPackageResolver.resolveVersionOfLinkedPackage(reportingProjectPackagePath, version.replace(LINKED_PACKAGE_PREFIX, ""));
+        }
+        return String.format("/%s@%s", name, version);
+    }
+
+    private Optional<NameVersion> parseNameVersionFromId(String id) {
+        // ids follow format: /name@version
+        try {
+            // It seems critical not to send this extra information in () or the kb will fail matching it.
+            if (id.contains("(")) {
+                id = id.split("\\(")[0];
+            }
+
+            int indexOfLastSlash = id.lastIndexOf("@");
+            String name = id.substring(1, indexOfLastSlash);
```

Comment:
Info come in the format of packageName@version(extra info). 
If we have the extra info we want to toss it. It doesn't help us find dependencies and it confuses the KB. Then we want to find the version based on @ instead of the v5 /